src/plugins/GUITestBase/src/GTUtilsMsaEditor.cpp

```diff
@@ -674,6 +677,38 @@ QListWidget* GTUtilsMsaEditor::getExcludeListWidget(HI::GUITestOpStatus& os) {
 }
 #undef GT_METHOD_NAME
 
+#define GT_METHOD_NAME "getMultilineMode"
+bool GTUtilsMsaEditor::getMultilineMode(HI::GUITestOpStatus &os)
+{
+    auto toolbar = GTToolbar::getToolbar(os, MWTOOLBAR_ACTIVEMDI);
+
+    // Get state of the "Multiline View" button on toolbar
+    auto mmode = GTToolbar::getToolButtonByAction(os, toolbar, "Multiline View");
+    CHECK_SET_ERR_RESULT(mmode != nullptr, "No \"Multiline View\" button", false);
+    return !mmode->isVisible() ? false : !mmode->isEnabled() ? false : mmode->isChecked();
+}
+#undef GT_METHOD_NAME
+
+#define GT_METHOD_NAME "setMultilineMode"
+void GTUtilsMsaEditor::setMultilineMode(HI::GUITestOpStatus &os, bool newMode)
+{
+    auto toolbar = GTToolbar::getToolbar(os, MWTOOLBAR_ACTIVEMDI);
+
+    // Press "Multiline View" button on toolbar
+    auto mmode = GTToolbar::getToolButtonByAction(os, toolbar, "Multiline View");
+    CHECK_SET_ERR_RESULT(mmode != nullptr, "No \"Multiline View\" button", );
+    bool oldMode = getMultilineMode(os);
+    if (oldMode == newMode) {
+        return;
+    }
+    CHECK_SET_ERR_RESULT(mmode->isVisible(), "\"Multiline View\" button is not visible", );
+    CHECK_SET_ERR_RESULT(mmode->isEnabled(), "\"Multiline View\" button is disabled", );
+    GTWidget::click(os, mmode);
+    GTUtilsTaskTreeView::waitTaskFinished(os);
```

Comment:
Do not put waitTaskFinished into utility methods. It waits for all tasks to be finished, even non-related ones.

Reply:
Hm, I will need to add the waitTaskFinished to the every call of this utility method